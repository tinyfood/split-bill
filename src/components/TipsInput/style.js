import styled from 'styled-components'

export const Tips = styled.div`
  display: flex;
  justify-content: space-between;
  margin-bottom: 24px;
`

export const Tip = styled.div`
  position: relative;
  width: 18%;
  padding: 7px 4px;
  text-align: center;
  color: #747d8c;
  border: 1.5px solid;
  border-color: ${props =>
    props.status === 'selected' ? '#2ecc71' : '#d1d8e0'}
  border-radius: 4px;
  transition: all .2s linear;
  user-select: none;
  cursor: pointer;
`

export const CheckMarkAnimated = styled.div`
  position: absolute;
  top: 0px;
  right: 0px;
  width: 35px;
  height: 15px;
  animation: zoomInFade 0.3s ease-in-out;
`
