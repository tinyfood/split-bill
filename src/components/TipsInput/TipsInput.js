import React, { Component } from 'react'
import { connect } from 'react-redux'
import { editTips } from '../../actions/split'
import { Tips, Tip, CheckMarkAnimated } from './style'
import { CheckMark } from '../../styles/icons/'

export class TipsInput extends Component {
  state = {
    tipsPercentages: [0, 10, 20, 30],
    currentTip: 0
  }

  handleTipClick = tipValue => {
    this.setState({
      currentTip: tipValue
    })

    this.props.editTips(tipValue)
  }

  render() {
    const { tipsPercentages, currentTip } = this.state

    return (
      <Tips>
        {tipsPercentages.map(tip => (
          <Tip
            key={tip}
            status={tip === currentTip ? 'selected' : 'not-selected'}
            onClick={this.handleTipClick.bind(this, tip)}
          >
            {tip}%
            {tip === currentTip && (
              <CheckMarkAnimated>
                <CheckMark />
              </CheckMarkAnimated>
            )}
          </Tip>
        ))}
      </Tips>
    )
  }
}

const mapDispatchToProps = dispatch => {
  return {
    editTips: tipValue => {
      dispatch(editTips(tipValue))
    }
  }
}

export default connect(
  null,
  mapDispatchToProps
)(TipsInput)
