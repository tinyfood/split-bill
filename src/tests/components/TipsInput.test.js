import React from 'react'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'
import { TipsInput } from '../../components/TipsInput/TipsInput'

test('should render TipsInput correctly', () => {
  const wrapper = shallow(<TipsInput />)
  expect(toJSON(wrapper)).toMatchSnapshot()
})
