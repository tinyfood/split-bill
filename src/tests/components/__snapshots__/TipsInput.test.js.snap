// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render TipsInput correctly 1`] = `
<styled.div>
  <styled.div
    key="0"
    onClick={[Function]}
    status="selected"
  >
    0
    %
    <styled.div>
      <CheckMark />
    </styled.div>
  </styled.div>
  <styled.div
    key="10"
    onClick={[Function]}
    status="not-selected"
  >
    10
    %
  </styled.div>
  <styled.div
    key="20"
    onClick={[Function]}
    status="not-selected"
  >
    20
    %
  </styled.div>
  <styled.div
    key="30"
    onClick={[Function]}
    status="not-selected"
  >
    30
    %
  </styled.div>
</styled.div>
`;
